apis/Google.Cloud.Logging.NLog/Google.Cloud.Logging.NLog/GoogleStackdriverTarget.cs

```diff
@@ -452,6 +492,87 @@ private LogEntry BuildLogEntry(LogEventInfo loggingEvent)
             return logEntry;
         }
 
+        private static Value BuildJsonValue(object objectValue)
+        {
+            var typeCode = Convert.GetTypeCode(objectValue);
+            switch (typeCode)
+            {
+                case TypeCode.Object:
+                    if (objectValue is System.Collections.IDictionary dictionary)
+                    {
+                        // Using IDictionaryEnumerator from IDictionary.GetEnumerator() as it provides key/value-pair
+                        var dictionaryStruct = new Struct();
+                        var itemEnumerator = dictionary.GetEnumerator();
+                        while (itemEnumerator.MoveNext())
+                        {
+                            var dictionaryKey = itemEnumerator.Key?.ToString();
+                            if (string.IsNullOrEmpty(dictionaryKey))
+                            {
+                                continue;
+                            }
+
+                            var itemTypeCode = Convert.GetTypeCode(itemEnumerator.Value);
+                            dictionaryStruct.Fields.Add(dictionaryKey, BuildSimpleJsonValue(itemTypeCode, itemEnumerator.Value));

```

Comment:
Sorry, one more thing...
I'm fairly sure that this call to `BuildSimpleJsonValue` should really call `BuildJsonValue`, to support nested 'complex' values?
If so, I think there is this occurrence, plus two more further down this method which need changing. And please add at least one nested complex structure to the test.
Hopefully this will be the last change required to this PR!

Reply:
I left out nested collections, as I then have to add support for cyclic object-references together with controlling max recursion level. I was actually hoping you would have jumped in and shown a fancy method in Protobuf that handles all this boilerplate-code out of the box.

Reply:
OK, let's leave it as-is for now. It can always be improved later. As far as I know there's no magic support for this, but I'll check.